integrations/amazon_bedrock/src/amazon_bedrock_haystack/generators/amazon_bedrock.py

```diff
@@ -0,0 +1,312 @@
+import json
+import logging
+import re
+from typing import Any, ClassVar, Dict, List, Optional, Type, Union
+
+import boto3
+from botocore.exceptions import BotoCoreError, ClientError
+from haystack import component, default_from_dict, default_to_dict
+
+from amazon_bedrock_haystack.errors import (
+    AmazonBedrockConfigurationError,
+    AmazonBedrockInferenceError,
+    AWSConfigurationError,
+)
+from amazon_bedrock_haystack.generators.amazon_bedrock_adapters import (
+    AI21LabsJurassic2Adapter,
+    AmazonTitanAdapter,
+    AnthropicClaudeAdapter,
+    BedrockModelAdapter,
+    CohereCommandAdapter,
+    MetaLlama2ChatAdapter,
+)
+from amazon_bedrock_haystack.generators.amazon_bedrock_handlers import (
+    DefaultPromptHandler,
+    DefaultTokenStreamingHandler,
+    TokenStreamingHandler,
+)
+
+logger = logging.getLogger(__name__)
+
+AWS_CONFIGURATION_KEYS = [
+    "aws_access_key_id",
+    "aws_secret_access_key",
+    "aws_session_token",
+    "aws_region_name",
+    "aws_profile_name",
+]
+
+
+@component
+class AmazonBedrockGenerator:
+    """
+    Generator based on a Hugging Face model.
+    This component provides an interface to generate text using a Hugging Face model that runs locally.
+
+    Usage example:
+    ```python
+    from amazon_bedrock_haystack.generators.amazon_bedrock import AmazonBedrockGenerator
+
+    generator = AmazonBedrockGenerator(
+        model_name="anthropic.claude-v2",
+        max_length=99,
+        aws_access_key_id="...",
+        aws_secret_access_key="...",
+        aws_session_token="...",
+        aws_profile_name="...",
+        aws_region_name="..."
+    )
+
+    print(generator.run("Who is the best American actor?"))
+    ```
+    """
+
+    SUPPORTED_MODEL_PATTERNS: ClassVar[Dict[str, Type[BedrockModelAdapter]]] = {
+        r"amazon.titan-text.*": AmazonTitanAdapter,
+        r"ai21.j2.*": AI21LabsJurassic2Adapter,
+        r"cohere.command.*": CohereCommandAdapter,
+        r"anthropic.claude.*": AnthropicClaudeAdapter,
+        r"meta.llama2.*": MetaLlama2ChatAdapter,
+    }
+
+    def __init__(
+        self,
+        model_name: str,
+        aws_access_key_id: Optional[str] = None,
+        aws_secret_access_key: Optional[str] = None,
+        aws_session_token: Optional[str] = None,
+        aws_region_name: Optional[str] = None,
+        aws_profile_name: Optional[str] = None,
+        max_length: Optional[int] = 100,
+        **kwargs,
+    ):
+        if model_name is None or len(model_name) == 0:
```

Comment:
I think we can simply check Truthy/Falsy and raise exception

Reply:
Done